Convert to DEB822 sources on upgrades to Groovy

isantop opened the pull request:
Adds a small conversion item in the postinst that will remove Ubuntu/Pop Sources from the sources.list file and add a default system.sources file in sources.list.d

Commit:
fix(postinst): upgrade traditional sources to Deb822-format sources

This only applies to the system software sources, and will leave existing
sources in sources.list intact.

Commit:
fix(postinst): also convert if using apt.pop-os.org

jackpot51 commented on debian/pop-default-settings.postinst:
```diff
@@ -99,8 +107,22 @@ case "$1" in
 
+        # Convert system sources from one-line format to DEB822, if present
+        # Generous matching, be more specific if found
+        if grep -e "ubuntu.com" -e "apt.pop-os.org/ubuntu" /etc/apt/sources.list 
```
You can add /ubuntu to all the ubuntu.com lines just like on apt.pop-os.org/ubuntu

isantop replied:
Updated.

Commit:
fix(postinst): address PR feedback

jackpot51 reviewed (dismissed):
I approve. But with things like this, it is good to make very certain there aren't false positives. So @pop-os/quality-assurance should take a look

Commit:
Merge branch 'master_groovy' into 822-upgrades_groovy

Commit:
fix(postinst): fix bad sed expression

Commit:
Merge branch '822-upgrades_groovy' of github.com:pop-os/default-settings into 822-upgrades_groovy

leviport reviewed (dismissed):
Seems to be working as expected after the sed fixes. Here are the contents of the resulting `sources.list` file on a fresh-ish Focal install that was upgraded to Groovy: 
```
# Ubuntu Repositories





# Pop!_OS Repositories

deb http://ppa.launchpad.net/system76/pop/ubuntu groovy main

deb http://apt.pop-os.org/proprietary groovy main
```

I also tested with a copy of my sources.list file and Jacob's sources.list file, and the Ubuntu sources were removed as expected. I can't think of any potential false positives, so I think this is good to go.

Commit:
fix(postinst): convert apt-proprietary source to DEB822 on upgrades

Similarly to the system sources, it looks through sources.list for the line
and if it finds it, removes it. The source is re-added by the existing code
lower down.

jacobgkau reviewed (approved):
One observation, this re-enables proprietary if it's been disabled in legacy format (e.g. via Repoman on 20.04). I'm considering that expected behavior because pop-upgrade already does the same thing, and this is only going to happen once, during the upgrade. (If/when pop-upgrade is updated to handle deb822 instead of wiping everything and populating the default sources.list in legacy format, this may be something to think through further.)
